bee-network/src/conns/manager.rs

```diff
@@ -0,0 +1,217 @@
+                        trace!("Already connected to {}", peer_id);
+                        NUM_LISTENER_EVENT_PROCESSING_ERRORS.fetch_add(1, Ordering::Relaxed);
+                        continue;
+                    }
+
+                    // Prevent accepting banned peers.
+                    if banned_peers.contains(&peer_id).await {
+                        trace!("Ignoring peer. Cause: {} is banned.", peer_id);
+                        NUM_LISTENER_EVENT_PROCESSING_ERRORS.fetch_add(1, Ordering::Relaxed);
+                        continue;
+                    }
+
+                    let peer_info = if let Ok(peer_info) = peers.get_info(&peer_id).await {
+                        // If we have this peer id in our peerlist (but are not already connected to it),
+                        // then we allow the connection.
+                        peer_info
+                    } else {
+                        let peer_info = PeerInfo {
+                            address: peer_address,
+                            alias: None,
+                            relation: PeerRelation::Unknown,
+                        };
+
+                        if peers
+                            .insert(peer_id.clone(), peer_info.clone(), PeerState::Disconnected)
+                            .await
+                            .is_err()
+                        {
+                            trace!("Ignoring peer. Cause: Denied by peerlist.");
+                            NUM_LISTENER_EVENT_PROCESSING_ERRORS.fetch_add(1, Ordering::Relaxed);
+                            continue;
+                        } else {
+                            // We also allow for a certain number of unknown peers.
+                            info!("Allowing connection to unknown peer {}.", peer_id.short(),);
+
+                            peer_info
+                        }
+                    };
+
+                    log_inbound_connection_success(&peer_id, &peer_info);
+
+                    if let Err(e) = super::upgrade_connection(
+                        peer_id,
+                        peer_info,
+                        muxer,
+                        Origin::Inbound,
+                        internal_event_sender.clone(),
+                    )
+                    .await
+                    {
+                        error!("Error occurred during upgrading the connection. Cause: {}", e);
+                        NUM_LISTENER_EVENT_PROCESSING_ERRORS.fetch_add(1, Ordering::Relaxed);
+                        continue;
+                    }
+                }
+            }
+
+            info!("Listener stopped.");
+        });
+
+        trace!("Connection Manager started.");
+
+        Ok(Self::default())
+    }
+
+    async fn stop(self, _node: &mut N) -> Result<(), Self::Error> {
+        info!("Stopping spawned tasks...");
+        Ok(())
+    }
+}
+
+fn log_inbound_connection_success(peer_id: &PeerId, peer_info: &PeerInfo) {
```

Comment:
Is this being used once ? If yes, I'd probably not do a function, ends up being more verbose than needed.